openedx_ledger/views.py

```diff
@@ -1,6 +1,103 @@
 """
 Views for the openedx_ledger app.
 """
-# from django.shortcuts import render
+import logging
 
-# Create your views here.
+from django.http import HttpResponseRedirect
+from django.shortcuts import render
+from django.urls import reverse
+from django.views.generic import View
+
+from openedx_ledger.api import reverse_full_transaction
+from openedx_ledger.models import Transaction
+from openedx_ledger.signals.signals import ADMIN_INVOKED_TRANSACTION_UNENROLL
+
+logger = logging.getLogger(__name__)
+
+
+# TODO: Should we add a base class to do auth checks?
+class UnenrollLearnersView(View):
+    """
+    Admin view for the canceling platform enrollments form.
+    """
+    template = "edx_ledger/admin/unenroll.html"
+
+    def get(self, request, transaction_id):
+        """
+        Handle GET request - render "Cancel transaction without refund" form.
+
+        Arguments:
+            request (django.http.request.HttpRequest): Request instance
+            transaction_uuid (str): Enterprise Customer UUID
+
+        Returns:
+            django.http.response.HttpResponse: HttpResponse
+        """
+        return render(
+            request,
+            self.template,
+            {'transaction': Transaction.objects.get(uuid=transaction_id)}
+        )
+
+    def post(self, request, transaction_id):
+        """
+        Handle POST request - handle form submissions.
+
+        Arguments:
+            request (django.http.request.HttpRequest): Request instance
+        """
+        logger.info(f"sending admin invoked transaction unenroll signal for transaction: {transaction_id}")
+        ADMIN_INVOKED_TRANSACTION_UNENROLL.send(
+            sender=self,
+            transaction=Transaction.objects.get(uuid=transaction_id),
+            admin=request.user,
+        )
+        url = reverse("admin:openedx_ledger_transaction_change", args=(transaction_id,))
+        return HttpResponseRedirect(url)
+
+
+class UnenrollLearnersWithRefundView(View):
+    """
+    Admin view for the canceling platform enrollments and reversing transaction objects form.
+    """
+    template = "edx_ledger/admin/unenroll_with_refund.html"
+
+    def get(self, request, transaction_id):
+        """
+        Handle GET request - render "Cancel transaction with refund" form.
+
+        Arguments:
+            request (django.http.request.HttpRequest): Request instance
+            transaction_uuid (str): Enterprise Customer UUID
+
+        Returns:
+            django.http.response.HttpResponse: HttpResponse
+        """
+        return render(
+            request,
+            self.template,
+            {'transaction': Transaction.objects.get(uuid=transaction_id)}
+        )
+
+    def post(self, request, transaction_id):
```

Comment:
Bunch of business logic below that should probably be in a shared utility function that can be re-used later for the `/api/v1/transactions/<>/reverse` REST API.

Reply:
this is no longer a concern with the current implementation